docs/entropy.md

```diff
@@ -0,0 +1,71 @@
+# Using the Firecracker entropy device
+
+## What is the entropy device
+
+An entropy device is a [`virtio-rng` device][1] that provides guests with
+"high-quality randomness for guest use". Guests issue requests in the form of a
+buffer that will be filled with random bytes from the device. The source of
+random bytes that the device will use to fill the buffers is an implementation
+decision.
+
+On the guest side, random bytes received by the device are used by the kernel
+as an extra source of entropy. Moreover, the guest VirtIO driver exposes the
+`/dev/hwrng` character device. User-space applications can use this device to
+request random bytes from the device.
+
+## Firecracker implementation
+
+Firecracker implements the `virtio-rng` device starting with version 1.4 as an
+optional device. Users can configure a single `virtio-rng` device, through the
+`/entropy` API endpoint. The request body includes a single (optional) rate
+limiter parameter.
+
+For example, users can configure the entropy device, with a bandwidth rate
+limiter of 10KB/sec like this:
+
+```console
+curl --unix-socket $socket_location -i \
+    -X PUT 'http://localhost/entropy' \
+    -H 'Accept: application/json' \
+    -H 'Content-Type: application/json' \
+    -d "{
+        \"rate_limiter\": {
+            \"bandwidth\": {
+                \"size\": 1000,
+                \"one_time_burst\": 0,
+                \"refill_time\": 100
+            }
+        }
+    }"
+```
+
+If a configuration file is used for configuring a microVM, the same setup can
+be achieved by adding a section like this:
+
+```json
+"entropy": {
+    "rate_limiter": {
+        "bandwidth" {
+            "size": 1000,
+            "one_time_burst": 0,
+            "refill_time": 100
+        }
+    }
+}
+```
+
+On the host side, the implementation on the host side is using [`OsRng`][2] to
+get random bytes from the host kernel. The [implementation][3] of `OsRng` on
+Linux uses the `getrandom(2)` system call when available, otherwise it falls
+back to `/dev/urandom` after successfully polling `/dev/random`.
```

Comment:
```suggestion
On the host side, firecracker uses [`OsRng`][2] to
get random bytes from the host kernel. The [implementation][3] of `OsRng` on
Linux uses the `getrandom(2)` system call when available, otherwise it falls
back to `/dev/urandom` after successfully polling `/dev/random`.
```